src/history/FileTransferInfo.h

```diff
@@ -36,21 +35,44 @@ class FileTransferInfo
     }
 
     FileTransferInfo(TmpDir const& snapDir, std::string const& snapType,
-                     uint32_t checkpointLedger)
+                     uint32_t checkpointLedger, bool cascade = false)
         : mType(snapType)
         , mHexDigits(fs::hexStr(checkpointLedger))
-        , mLocalPath(snapDir.getName() + "/" + baseName_nogz())
+        , mLocalPath(getLocalDir(snapDir, snapType, cascade) + "/" +
+                     baseName_nogz())
     {
     }
 
     FileTransferInfo(TmpDir const& snapDir, std::string const& snapType,
-                     std::string const& hexDigits)
+                     std::string const& hexDigits, bool cascade = false)
         : mType(snapType)
         , mHexDigits(hexDigits)
-        , mLocalPath(snapDir.getName() + "/" + baseName_nogz())
+        , mLocalPath(getLocalDir(snapDir, snapType, cascade) + "/" +
+                     baseName_nogz())
     {
     }
 
+    std::string
+    getLocalDir(TmpDir const& snapDir, std::string const& snapType,
+                bool cascade) const
+    {
+        auto localDir = snapDir.getName();
+        if (cascade)
+        {
+            localDir += "/" + fs::remoteDir(snapType, mHexDigits);
+            int retries = 5;
```

Comment:
comment on why you need retries, I imagine it's because of race conditions? Why 5?
Should it sleep a little between retries?

Reply:
this is similar to what we do with TmpDir, i imagine there could be OS-related errors, hence the retries. Ok, sleeping in between makes sense.